code/game/turfs/open/space/transit.dm

```diff
@@ -40,12 +40,43 @@
 	var/max = world.maxx-TRANSITIONEDGE
 	var/min = 1+TRANSITIONEDGE
 
-	var/list/possible_transtitons = list()
-	for(var/A in SSmapping.z_list)
-		var/datum/space_level/D = A
-		if (D.linkage == CROSSLINKED)
-			possible_transtitons += D.z_value
-	var/_z = pick(possible_transtitons)
+	//Find our location
+	var/_z = 2
+
+	var/should_make_level = ismob(AM)
+	if(isobj(AM))
```

Comment:
```suggestion
	if(!should_make_level || isobj(AM))
```

Reply:
This code wouldn't work, it would need to be
```
if(!should_make_level && isobj(AM))
```